pinot-spi/src/main/java/org/apache/pinot/spi/config/table/JsonIndexConfig.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.table;
+
+import com.google.common.base.Preconditions;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.pinot.spi.config.BaseJsonConfig;
+
+
+public class JsonIndexConfig extends BaseJsonConfig {
+  private int _maxLevels = -1;
+  private boolean _excludeArray = false;
+  private boolean _disableCrossArrayUnnest = false;
+  private Set<String> _includePaths;
+  private Set<String> _excludePaths;
+  private Set<String> _excludeFields;
```

Comment:
would it make sense to add an "_excludeTypes"? 
for example, if the node is an array i will not need to enumerate them all out

also good to add a javadoc to explain which overwrites which (e.g. in both include and exclude, etc)?

Reply:
`excludeArray` is for that purpose. IMO, `excludeValue` and `excludeObject` doesn't make a lot of sense for JSON.
Added javadoc to explain the behavior of each config.